lib/js/caml_array.js

```diff
@@ -61,6 +62,31 @@ function caml_array_concat(l) {
   return result;
 }
 
+function caml_array_set(xs, index, newval) {
+  if (index < 0 || index >= xs.length) {
+    throw [
```

Comment:
@dorafmon would you create a following PR to lift such allocation into the toplevel.  such as 

```ocaml
let outbound_exn = Invalid_argument "out of bounds"
let caml_array_set xs index new val = 
   ...
```